File: data/gpt-4o/readable_format_HPEC/cpp/file232.cpp
void init_array(int ni, int nj, double alpha, double beta, double c[100][100], double a[100][100], double b[100][100])
{
 alpha = 32412.0;
 beta = 2123.0;

 for(int i = 0; i < ni; i++)
 for(int j = 0; j < nj; j++)
 {
 a[j][i] = (double(i) * double(j)) / double(ni);
 b[j][i] = (double(i) * double(j)) / double(ni);
 }

 for(int i = 0; i < ni; i++)
 for(int j = 0; j < ni; j++)
 {
 c[j][i] = (double(i) * double(j)) / double(ni);
 }
}

File: data/CodeLlama/readable_format_HPEC/fortran/file232.f90
subroutine init_array(ni, nj, alpha, beta, c, a, b)
implicit none

DATA_TYPE, dimension(nj, ni) :: a
DATA_TYPE, dimension(nj, ni) :: b
DATA_TYPE, dimension(ni, ni) :: c
DATA_TYPE :: alpha, beta
integer :: ni, nj
integer :: i, j

alpha = 32412.0D0
beta = 2123.0D0

do i = 1, ni
do j = 1, nj
a(j, i) = (DBLE(i - 1) * DBLE(j - 1)) / DBLE(ni)
b(j, i) = ((DBLE(i - 1) * DBLE(j - 1))) / DBLE(ni)
end do
end do
do i = 1, ni
do j = 1, ni
c(j, i) = ((DBLE(i - 1) * DBLE(j - 1))) / DBLE(ni)
end do
end do
end subroutine
